docs/compilers/CSharp/Warnversion Warning Waves.md

```diff
@@ -0,0 +1,25 @@
+﻿# /warnversion warning "waves"
+
+The C# compiler flag `/warnversion` controls optional warnings.
+When we introduce new warnings that can be reported on existing code,
+we do so under an opt-in system so that programmers do not see new warnings
+without taking action to enable them.
+For that purpose, we have introduced the compiler flag "`/warnversion=n`"
+where `n` is a whole number or a decimal number.
+For a warning that was introduced in dotnet version `k`,
+that warning will be produced if the warning version `n` specified when compiling is
+greater than or equal to `k` and a compiler shipped with dotnet version
+`k` or later is used to compile the code.
+
+The default warning version is `0` (produce no optional warnings).
+Our first warning under control of `/warnversion` was introduced in version `5`
+as part of .NET 5.
+If you want the compiler to produce all applicable warnings, you can specify
+`/warnversion=9999`.
```

Comment:
Should this instead be a special string, say `latest` instead of a number? This would also match the `LangVersion` property values.

Reply:
Open issue tracked by [https://github.com/dotnet/roslyn/issues/45941](https://github.com/dotnet/roslyn/issues/45941)

---
In reply to: [453250389](https://github.com/dotnet/roslyn/pull/45750#discussion_r453250389) [](ancestors = 453250389)